drivers/Watchdog.cpp

```diff
@@ -18,48 +18,83 @@
 
 #include "Watchdog.h"
 
-namespace mbed {
-
-watchdog_status_t Watchdog::start(const uint32_t timeout)
+namespace mbed
 {
-    if (timeout == 0) {
-        return WATCHDOG_STATUS_INVALID_ARGUMENT;
-    }
 
-    if (timeout > max_timeout()) {
-        return WATCHDOG_STATUS_INVALID_ARGUMENT;
-    }
+Watchdog *Watchdog::first;
 
-    watchdog_config_t config;
-    config.timeout_ms = timeout;
+void Watchdog::add_to_list()
+{
+    this->next = first;
```

Comment:
May want to skip adding it to the list if `_is_initialized` is set, as this means you are already in the list.